Replace TargetType validate: Bool with validationType: ValidationType

SD10 opened the pull request:
This resolves #1447 and completes #1454 which was started by @amaurydavid

It replaces the `Boolean` property of `TargetType` with a `ValidationType` enum allowing the user to customize the status codes used by Alamofire's `validate(statusCodes:)` method.

TODO:
- [x] CHANGELOG
- [x] Update Documentation
- [x] Update Example app

SD10 commented on Tests/MoyaProviderIntegrationTests.swift:
```diff
@@ -255,8 +255,9 @@ class MoyaProviderIntegrationTests: QuickSpec {
                         expect(log).to( contain("Response:") )
-                        expect(log).to( contain("{ URL: https://api.github.com/zen } { status code: 200, headers") )
-                        expect(log).to( contain("\"Content-Length\" = 43;") )
+                        expect(log).to( contain("{ URL: https://api.github.com/zen } { Status Code: 200, Headers") )
```
Looks like this is the cause of CI failure now 🤔

SD10 replied:
An update on this:
My tests won't pass locally without these changes, but the changes cause CI to fail.

SD10 commented:
@sunshinejr will update the docs tonight and start a `Moya 12.0` migration guide. I'm not sure about the CI error, right now that specific test won't pass locally but CI runs fine. I guess I'll just use CI to run my tests for now. Curious if others face this issue on their local copy as well

sunshinejr commented:
@SD10 which test didn't pass locally for you? There are few tests that should be separated, I talk about it at #1424, basically some of the test fail under bad internet connection.

SD10 commented on Tests/MoyaProviderIntegrationTests.swift:
```diff
@@ -255,8 +255,9 @@ class MoyaProviderIntegrationTests: QuickSpec {
                         expect(log).to( contain("Request Headers: [:]") )
                         expect(log).to( contain("HTTP Request Method: GET") )
                         expect(log).to( contain("Response:") )
-                        expect(log).to( contain("{ URL: https://api.github.com/zen } { status code: 200, headers") )
```
@sunshinejr This is the test that fails locally for me. The "status code: 200, headers" comes back with capitalized words when I run them locally and on CI they're not capitalized.

SD10 commented on Tests/MoyaProviderIntegrationTests.swift:
```diff
@@ -255,8 +255,9 @@ class MoyaProviderIntegrationTests: QuickSpec {
                         expect(log).to( contain("HTTP Request Method: GET") )
                         expect(log).to( contain("Response:") )
-                        expect(log).to( contain("{ URL: https://api.github.com/zen } { status code: 200, headers") )
-                        expect(log).to( contain("\"Content-Length\" = 43;") )
```
Locally, 43 comes back as (43) wrapped in parentheses. On CI it does not 😓

SD10 commented:
Ok I've completed the following:
- [X] Updated the example app
- [X] Added a deprecation warning for `validate: Bool` because I noticed this change fails silently
- [X] Updated the docs, only `AlamofireValidation.md` seemed to be affected

SD10 commented:
I'm tagging #1357 because `AlamofireValidation.md` will need to be updated in the Chinese documentation.
